### PR TITLE
go1.19 support 

### LINKED ISSUE #2922: go-critic: checkers.init panic with go1.19beta1
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc).
- [X] Yes, I've tried with the standalone linter if available. (https://golangci-lint.run/usage/linters/)

### Description of the problem

When I try to use golangci-lint with go 1.19beta1 in GHA CI, I get:
```
Running [/home/runner/golangci-lint-1.46.2-linux-amd64/golangci-lint run --out-format=github-actions] in [] ...
panic: load embedded ruleguard rules: rules/rules.go:13: can't load fmt
  
  goroutine 1 [running]:
  github.com/go-critic/go-critic/checkers.init.22()
  	github.com/go-critic/go-critic@v0.6.3/checkers/embedded_rules.go:47 +0x4b4
```

### Version of golangci-lint

1.46.2

### Configuration file

not relevant

### Go environment

<details>

```console
$ go env
# paste output here
```
GO111MODULE=""
  GOARCH="amd64"
  GOBIN=""
  GOCACHE="/home/runner/.cache/go-build"
  GOENV="/home/runner/.config/go/env"
  GOEXE=""
  GOEXPERIMENT=""
  GOFLAGS=""
  GOHOSTARCH="amd64"
  GOHOSTOS="linux"
  GOINSECURE=""
  GOMODCACHE="/home/runner/go/pkg/mod"
  GONOPROXY=""
  GONOSUMDB=""
  GOOS="linux"
  GOPATH="/home/runner/go"
  GOPRIVATE=""
  GOPROXY="https://proxy.golang.org,direct"
  GOROOT="/opt/hostedtoolcache/go/1.19.0-beta.1/x64"
  GOSUMDB="sum.golang.org"
  GOTMPDIR=""
  GOTOOLDIR="/opt/hostedtoolcache/go/1.19.0-beta.1/x64/pkg/tool/linux_amd64"
  GOVCS=""
  GOVERSION="go1.19beta1"
  GCCGO="gccgo"
  GOAMD64="v1"
  AR="ar"
  CC="gcc"
  CXX="g++"
  CGO_ENABLED="1"
  GOMOD="/home/runner/work/runc/runc/go.mod"
  GOWORK=""
  CGO_CFLAGS="-g -O2"
  CGO_CPPFLAGS=""
  CGO_CXXFLAGS="-g -O2"
  CGO_FFLAGS="-g -O2"
  CGO_LDFLAGS="-g -O2"
  PKG_CONFIG="pkg-config"
  GOGCCFLAGS="-fPIC -m64 -pthread -Wl,--no-gc-sections -fmessage-length=0 -fdebug-prefix-map=/tmp/go-build51[30](https://github.com/opencontainers/runc/runs/6923780174?check_suite_focus=true#step:3:31)76706=/tmp/go-build -gno-record-gcc-switches"
</details>

### Verbose output of running

n/a

### Code example or link to a public repository

See https://github.com/opencontainers/runc/runs/6923780174?check_suite_focus=true

https://github.com/opencontainers/runc/pull/3514

### PR DESCRIPTION
**Pull request by ldez**
This PR is to evaluate and prepare golangci-lint to [go1.19](https://tip.golang.org/doc/go1.19).

This PR will evolve during the beta and rc phases of [go1.19](https://tip.golang.org/doc/go1.19).

Fixes #2922

### CODE REVIEW

File: .github/workflows/post-release.yml [hunk 1]
```diff
@@ -14,8 +14,8 @@ jobs:
       - name: Install Go
         uses: actions/setup-go@v3
         with:
-          # stable: 'false' # Keep this line to be able to use rc and beta version of Go (ex: 1.18.0-rc1).
-          go-version: 1.18
+          # stable: 'false' # Keep this line to be able to use rc and beta version of Go (ex: 1.18.0-rc.1).
```
Comment (lopezator) [inline .github/workflows/post-release.yml:17]: Sorry, I'm trying to help but don't have much context, why the "." Is added? I see the former format in the go repo tags. Furthermore, should we take the opportunity to update the comment to 1.19 or do we have to keep it in 1.18 for some reason?

Comment (ldez) [reply .github/workflows/post-release.yml:17]: It's related to the [setup-go](https://github.com/actions/setup-go)

https://github.com/actions/setup-go#supported-version-syntax

The message is just for the pattern.

Comment (ldez) [reply .github/workflows/post-release.yml]: I improved the comments.

**Review by lafriks — APPROVED** [review]

**Comment by yermulnik** [comment]
```shell
> go version
go version go1.19 linux/amd64

> golangci-lint
panic: load embedded ruleguard rules: rules/rules.go:13: can't load fmt

goroutine 1 [running]:
github.com/go-critic/go-critic/checkers.init.22()
        github.com/go-critic/go-critic@v0.6.3/checkers/embedded_rules.go:47 +0x4b4
```
or should this be reported into https://github.com/go-critic/go-critic ?

**Comment by invidian** [comment]
https://github.com/golangci/golangci-lint/commit/1557692e59c0db1b6eaa771a5887f9434158683d has not been released yet.

**Comment by Don-Vito** [comment]
> ```shell
> > go version
> go version go1.19 linux/amd64
> 
> > golangci-lint
> panic: load embedded ruleguard rules: rules/rules.go:13: can't load fmt
> 
> goroutine 1 [running]:
> github.com/go-critic/go-critic/checkers.init.22()
>         github.com/go-critic/go-critic@v0.6.3/checkers/embedded_rules.go:47 +0x4b4
> ```
> 
> or should this be reported into https://github.com/go-critic/go-critic ?

Looks duplicate to https://github.com/golangci/golangci-lint/issues/3053

**Comment by ldez** [comment]
We will create a release with go1.19 support at 6:00 pm (UTC)

**Comment by yermulnik** [comment]
> Looks duplicate to #3053

#3053 cycles back to here =)

> We will create a release with go1.19 support at 6:00 pm (UTC)

Thanks 👍🏻

**Comment by yermulnik** [comment]
I still can see the panic:
```shell
> brew info golangci-lint
golangci-lint: stable 1.48.0 (bottled), HEAD
Fast linters runner for Go
https://golangci-lint.run/
/home/linuxbrew/.linuxbrew/Cellar/golangci-lint/1.48.0 (9 files, 29.5MB) *
  Poured from bottle on 2022-08-05 at 13:03:48
From: https://github.com/Homebrew/homebrew-core/blob/HEAD/Formula/golangci-lint.rb
License: GPL-3.0-only
==> Dependencies
Required: go ✔
==> Options
--HEAD
        Install HEAD version
==> Caveats
Bash completion has been installed to:
  /home/linuxbrew/.linuxbrew/etc/bash_completion.d
==> Analytics
install: 16,649 (30 days), 39,753 (90 days), 109,201 (365 days)
install-on-request: 16,638 (30 days), 39,726 (90 days), 109,094 (365 days)
build-error: 9 (30 days)

> golangci-lint
panic: load embedded ruleguard rules: rules/rules.go:13: can't load fmt

goroutine 1 [running]:
github.com/go-critic/go-critic/checkers.init.22()
        github.com/go-critic/go-critic@v0.6.3/checkers/embedded_rules.go:47 +0x4b4
```

**Comment by ldez** [comment]
https://github.com/golangci/golangci-lint/issues/2673#issuecomment-1205916993

**Pull request merged**